benchmarks/actors/src/main/scala/org/renaissance/actors/AkkaUct.scala

```diff
@@ -2,22 +2,21 @@ package org.renaissance.actors
 
 import edu.rice.habanero.actors.AkkaActorState
 import edu.rice.habanero.benchmarks.uct.UctAkkaActorBenchmark
-import org.renaissance.Config
-import org.renaissance.License
-import org.renaissance.RenaissanceBenchmark
+import org.renaissance.Benchmark._
+import org.renaissance.{Config, License, RenaissanceBenchmark}
 
+@Summary("Runs the Unbalanced Cobwebbed Tree actor workload in Akka.")
+@Licenses(Array(License.MIT))
+@Repetitions(24)
 class AkkaUct extends RenaissanceBenchmark {
-  override def description(): String =
-    "Runs the Unbalanced Cobwebbed Tree actor workload in Akka."
 
-  override def licenses(): Array[License] = License.create(License.MIT)
+  // TODO: Consolidate benchmark parameters across the suite.
+  //  See: https://github.com/renaissance-benchmarks/renaissance/issues/27
 
-  override def defaultRepetitions = 24
+  private var numIterations: Int = 10
```

Comment:
Am I missing something if there is 24 in `@Repetitions`?

Reply:
> 
> > A slew of changes to simplify bits of build and simplify things in RenaissanceSuite. For example, the build does not generate the benchmark-group.txt file, and the harness tries to use benchmark-details.properties for most things.
> 
> I was wondering whether it would make more sense to have everything in something like `META-INF/RENAISSANCE` in `MANIFEST.MF`-style format?

That's probably doable, especially if we can put all metadata into a single file. Currently there are two.

> 
> > Add Benchmark interface with annotation interfaces for metadata.
> 
> I believe text files next to the benchmarks would work better (e.g. for web-site generation) but definitely this is better than normal methods.

I understand. In the end, I wanted to avoid potential benchmark authors to deal with description files for this kind of information (workload sizes are a different matter). I take it that we can generate any files necessary for website generation during build.

Reply:
> Am I missing something if there is 24 in `@Repetitions`?

numIterations is an internal loop counter, while repetitions is the external one.

Reply:
> I take it that we can generate any files necessary for website generation during build.

My line of thought was that it would be nice if the website would be more independent of the *compilation* of the suite. That way, the website would not need to be mirrored under the suite repo and Jekyll could get list of benchmarks etc. without need to compile anything.



> > Am I missing something if there is 24 in `@Repetitions`?
> 
> numIterations is an internal loop counter, while repetitions is the external one.

My bad, sorry.

Reply:
> > I take it that we can generate any files necessary for website generation during build.
> 
> My line of thought was that it would be nice if the website would be more independent of the _compilation_ of the suite. That way, the website would not need to be mirrored under the suite repo and Jekyll could get list of benchmarks etc. without need to compile anything.

We could keep such a generated file under version control and push it to (potentially independent) website repo on demand.

After some thought, it seemed better to me if a benchmark author (human) can have the metadata (including descriptions) next to the implementation class (much of it will not change but the descriptions/repetitions even name/group -- they can be set using annotations as well, the names are automatically derived if the annotations are unset).

We can accommodate the website generators (tools) in a different way.

Reply:
> > > I take it that we can generate any files necessary for website generation during build.
> > 
> > My line of thought was that it would be nice if the website would be more independent of the _compilation_ of the suite. That way, the website would not need to be mirrored under the suite repo and Jekyll could get list of benchmarks etc. without need to compile anything.
> 
> We could keep such a generated file under version control and push it to (potentially independent) website repo on demand.
> 
> After some thought, it seemed better to me if a benchmark author (human) can have the metadata (including descriptions) next to the implementation class (much of it will not change but the descriptions/repetitions even name/group -- they can be set using annotations as well, the names are automatically derived if the annotations are unset).
> 
> We can accommodate the website generators (tools) in a different way.

That makes sense. If we would dump all this information into YAML (or something similar that Jekyll [can read](https://jekyllrb.com/docs/datafiles/)) and version it, it should work too.

Reply:
Sounds like a plan to me and it should be doable from the build system.

Reply:
> > > I take it that we can generate any files necessary for website generation during build.
> > 
> > 
> > My line of thought was that it would be nice if the website would be more independent of the _compilation_ of the suite. That way, the website would not need to be mirrored under the suite repo and Jekyll could get list of benchmarks etc. without need to compile anything.
> 
> We could keep such a generated file under version control and push it to (potentially independent) website repo on demand.
> 
> After some thought, it seemed better to me if a benchmark author (human) can have the metadata (including descriptions) next to the implementation class (much of it will not change but the descriptions/repetitions even name/group may -- they can be set using annotations as well, the names are automatically derived if the annotations are unset).
> 
> We can accommodate the website generators (tools) in a different way.

I like the sentiment that the authors should have all the metadata close to the implementation, because it decreases the likelihood that they miss an update.